.github/pull_request_template.md

```diff
@@ -0,0 +1,50 @@
+## Description
+
+Replace the content in this section with:
+- The motivation and context for this change (if it is not immediately clear from the title)
+- If it fixes an open issue, specify the issue number (e.g., "fixes #XXXX")
+- A summary of the behavior expected from this change
+- A description of tests performed
+
+## Author Progress Checklist:
+
+- [ ] Open draft pull request
+    - [ ] Make title clearly understandable in a standalone change log context
+    - [ ] Assign yourself the issue
+    - [ ] Add at least one label
+- [ ] Make code changes (if you haven't already)
+- [ ] Self-review of code
+    - [ ] My code follows the style guidelines of this project
+    - [ ] I have added comments to my code, particularly in hard-to-understand areas
+    - [ ] I have only committed the necessary changes for this fix or feature
+    - [ ] I have made corresponding changes to the documentation
+    - [ ] My changes generate no new warnings
+    - [ ] I have ensured that my fix is effective or that my feature works as intended by:
+        - [ ] exercising the code changes in the test framework, and/or
+        - [ ] manually verifying the changes (as explained in the the pull request description above)
+    - [ ] My changes pass all local tests
+    - [ ] My changes successfully passes CI checks
+    - [ ] Add any unit test for proof and documentation.
+- [ ] Move pull request out of draft mode and assign reviewers
+- [ ] Iterate with reviewers until all changes are approved
+    - [ ] Make changes in response to reviewer comments
+    - [ ] Re-request review in GitHub
+    - [ ] Add Chip for final review.
+
+## Reviewer Checklist:
+
+ - [ ] Read the pull request description
+ - [ ] Perform a code review on GitHub
+ - [ ] Confirm all CI checks pass and there are no build warnings
+ - [ ] Pull, build, and run automated tests locally
+ - [ ] Perform manual tests of the fix or feature locally
+ - [ ] Add any review comments, if applicable
+ - [ ] Submit review in GitHub as either
+     - [ ] Request changes, or
+     - [ ] Approve
+ - [ ] Iterate with author until all changes are approved
+
+ ## Final Check
```

Comment:
I don't think the "Final check" is necessary if we move the merge of the main branch as recommended.